extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -73,17 +78,28 @@ public void handle(RoutingContext event) {
                 //if proactive auth is used this is the only one
                 //if using lazy auth this can be modified downstream, to control authentication behaviour
                 event.put(QuarkusHttpUser.AUTH_FAILURE_HANDLER, new BiConsumer<RoutingContext, Throwable>() {
```

Comment:
Why can't RR just set it's own `AUTH_FAILURE_HANDLER`? This use case appears to be what it was designed for?

Reply:
Originally I did that, but sometimes response didn't end. This way, we are sure RR context is running so that exception is handled by them. I think I remember which tests where failing if that's important for you.

Reply:
I think `io.quarkus.jwt.test.RolesAllowedPostTest#postTextInvalidToken` and `io.quarkus.jwt.test.RolesAllowedPostTest#postTextWrongRole`. This way just feels safer.

Reply:
@stuartwdouglas Isn't there going to be a lot of duplication logic etc.? Only difference is what happens when we want to end it. But I think I could use inheritance so that only difference between the handlers is going to be how it ends. Good idea, I'll try it tomorrow.

Reply:
Thank you

Reply:
RESTEasy classic just removes it: https://github.com/stuartwdouglas/quarkus/blob/b76e53e23b60b48e9ca45ab6ea5e772e1186cc32/extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxRequestHandler.java#L119

Reply:
Sure, I checked and all the exceptions handled by the default auth failure handler are handled same by provided exception mappers. You were right, thanks!